### DISCUSSION TITLE
How can update comment list each time the page loads without getServerSideProps

### DISCUSSION THREAD

**Discussion by trangchongcheng** [vercel/next.js#28380] [Help]
Currently, i buiding a blog by nextjs and i have a problem haven't found a solution yet so can you suggest me some solutions.
I building a blog like http://dev.to. I have router for post like /post/[slug].tsx.
Im using getStacticProps for all post and every the post user can leave a comment like this comment.
When user add comment, if refresh the post then will see comment not show because the post had get from cache before.
At second refresh it get updated cause i set revalidate 1.getServerSideProps can fix it but i think it not good for this case.All post should cache to fast load.

Anyone help me? thank you so much

**Comment by oliverloops** [comment]
Hi @trangchongcheng, I understand you want to keep your site with the most statically possible approach. You mentioned you're using Incremental Static Regeneration with `revalidate: 1` which is a good approach for cached data but even with this a user is obtaining staled data and I believe you want to create a kind of real-time comment update. One idea that comes to my mind is try to re-fetch your data only on the client side (more specifically just in your comment section component and not the whole page) every few seconds, you can use something like `refetchInterval` approach from React Query library or maybe using native `fetch` and `setInterval` combined with some of the stale-while-revalidate features from SWR library.

If this helps, please let me know.

**Reply by trangchongcheng** [reply]
Thank to @oliverloops , it help me.I have switch get comments to client-side rendering with SWR library of vercel.Thank for giving me the solution :*

**Reply by oliverloops** [reply]
I'm glad this helped you!
